buffer/src/main/java/io/netty/buffer/PooledUnsafeHeapByteBuf.java

```diff
@@ -129,6 +129,34 @@ protected void _setLongLE(int index, long value) {
     }
 
     @Override
+    public ByteBuf setZero(int index, int length) {
+        if (PlatformDependent.javaVersion() >= 7) {
```

Comment:
@nmittler actually I think it makes more sense to keep it here. Let me just pull this in as it is